components/match2/wikis/teamfortress/match_group_input_custom.lua

```diff
@@ -36,231 +29,134 @@ local LINK_PREFIXES = {
 }
 
 -- containers for process helper functions
-local matchFunctions = {}
-local mapFunctions = {}
+local MatchFunctions = {}
+local MapFunctions = {}
 
 local CustomMatchGroupInput = {}
 
--- called from Module:MatchGroup
 ---@param match table
+---@param options table?
 ---@return table
-function CustomMatchGroupInput.processMatch(match)
-	-- Count number of maps, check for empty maps to remove, and automatically count score
-	match = matchFunctions.getBestOf(match)
-	match = matchFunctions.getScoreFromMapWinners(match)
+function CustomMatchGroupInput.processMatch(match, options)
+	local finishedInput = match.finished --[[@as string?]]
+	local winnerInput = match.winner --[[@as string?]]
 
-	-- process match
 	Table.mergeInto(match, MatchGroupInput.readDate(match.date))
-	match = matchFunctions.getOpponents(match)
-	match = matchFunctions.getTournamentVars(match)
-	return matchFunctions.getVodStuff(match)
-end
 
--- called from Module:Match/Subobjects
----@param map table
----@return table
-function CustomMatchGroupInput.processMap(map)
-	map = mapFunctions.getExtraData(map)
-	return mapFunctions.getScoresAndWinner(map)
-end
+	local opponents = Array.mapIndexes(function(opponentIndex)
+		return MatchGroupInput.readOpponent(match, opponentIndex, {})
+	end)
+	local games = CustomMatchGroupInput.extractMaps(match, #opponents)
+	match.bestof = MatchFunctions.getBestOf(match)
```

Comment:
The best of calculation is completley different now, is that intended?

Reply:
Is it possible to adjust from here to keep the ability for winner to override when a tied series has a winner?